# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TreeOperation.cpp
# Commit: 77388b0795b4
# Full Hash: 77388b0795b4f9a30171c922d6b649c52e5f6c87
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-23 09:34:36
# Description:
#   Bug 1745142 - Communicate encoding commitment via speculative load queue. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133996
# ==============================================================================

diff -r a388127095db -r 77388b0795b4 parser/html/nsHtml5TreeOperation.cpp
--- a/parser/html/nsHtml5TreeOperation.cpp	Wed Dec 22 15:47:08 2021 +0000
+++ b/parser/html/nsHtml5TreeOperation.cpp	Wed Dec 22 15:54:49 2021 +0000
@@ -148,7 +148,7 @@
 
     void operator()(const opDoneCreatingElement& aOperation) {}
 
-    void operator()(const opSetDocumentCharset& aOperation) {}
+    void operator()(const opUpdateCharsetSource& aOperation) {}
 
     void operator()(const opCharsetSwitchTo& aOperation) {}
 
@@ -980,10 +980,8 @@
       return NS_OK;
     }
 
-    nsresult operator()(const opSetDocumentCharset& aOperation) {
-      auto encoding = WrapNotNull(aOperation.mEncoding);
-      mBuilder->SetDocumentCharsetAndSource(encoding,
-                                            aOperation.mCharsetSource);
+    nsresult operator()(const opUpdateCharsetSource& aOperation) {
+      mBuilder->UpdateCharsetSource(aOperation.mCharsetSource);
       return NS_OK;
     }
 